security/advancedtls/advancedtls_test.go

```diff
@@ -412,6 +426,11 @@ func TestClientServerHandshake(t *testing.T) {
 				VerifyPeer:        test.serverVerifyFunc,
 				VType:             test.serverVType,
 			}
+			servertConfig, _ := serverOptions.config()
```

Comment:
Calling an extra config() function doesn't look ideal to me, because in `NewServerCreds`,  config() will be called. 
It gives me a sense that we might want to have a separate tests for this behavior only. Can we have a separate test with the name "TestSystemRootCertificates", with the relevant changes?